src/screens/NewTerm.js

```diff
@@ -0,0 +1,201 @@
+import LanguageSwitcher from '../components/LanguageSwitcher/LanguageSwitcher';
+import BackIcon from '../components/ToolButton/assets/back.svg';
+import ModeIcon from '../components/ToolButton/assets/mode.svg';
+
+import {LANGUAGE_NAMES, getAllCategories} from '../data/dataUtils';
+import SectionHeading from '../components/SectionHeading/SectionHeading';
+
+export default ({
+  //nav provider
+  navigation,
+  categories,
+  setCategories,
+  addNewPhrases,
+}) => {
+  const [selectedCategory, setSelectedCategory] = useState('');
+  const [malagasyText, setMalagasyText] = useState('');
+  const [englishText, setEnglishText] = useState('');
+  const [categoryId, setCategoryId] = useState('');
+
+  // find the selected category
+  const findSelectedId = Text => {
+    setSelectedCategory(Text);
+    const selectedCat = categories.find(cat => cat.name.en === Text);
+    setCategoryId(selectedCat.id);
+  };
+  const addnewPhraseForCategory = () => {
+    const newPhrase = {
+      catId: categoryId,
+      id: uuid(),
+      name: {en: englishText, mg: malagasyText},
+    };
+    addNewPhrases(newPhrase);
+    // clear input
+    setMalagasyText('');
+    setEnglishText('');
+  };
+
+  useEffect(() => {
+    // fetch categories
+    const categories = getAllCategories();
+    setCategories(categories);
+  }, []);
+
+  return (
+    <SafeAreaView style={{flex: 1}}>
+      <KeyboardAvoidingView style={{flex: 1}} behavior="padding">
+        <View style={{paddingHorizontal: 35, paddingVertical: 23}}>
+          <View style={styles.header}>
+            <ToolBar
+              button={
+                <ToolButton
+                  onPress={() => {
+                    navigation.navigate('Home');
+                  }}>
+                  <BackIcon width={24} height={24} fill="#FFFFFF" />
+                </ToolButton>
+              }
+            />
+            <ToolBar
+              button={
+                <LanguageSwitcher
+                  firstLanguage={LANGUAGE_NAMES.EN}
+                  LeftText="EN"
+                  RightText="MA"
+                  color="#FFFFFF"
+                  iconType=""
+                  iconName="swap-horiz"
+                  onPress={() => null}
+                  iconSize={24}
+                />
+              }
+            />
+            <ToolBar
+              button={
+                <ToolButton onPress={action('clicked-add-button')}>
+                  <ModeIcon width={24} height={24} fill="#FFFFFF" />
+                </ToolButton>
+              }
+            />
+          </View>
+          <View style={styles.headerCategoryWrapper}>
+            <SectionHeading text="Category: " />
+            <View>
+              <Picker
+                selectedValue={selectedCategory}
+                style={{
+                  width: 300,
+                  height: 100,
+                }}
+                mode="dropdown"
+                onValueChange={Text => findSelectedId(Text)}>
```

Comment:
This looks fine but it would be better if you use another more descriptive name for the function "findSelectedId" in line 108 because this function mainly is selectsing a category from the dropdown but not only finding an id.